presto-main/src/main/java/io/prestosql/sql/analyzer/QueryExplainer.java

```diff
@@ -175,7 +176,7 @@ public Plan getLogicalPlan(Session session, Statement statement, List<Expression
         PlanNodeIdAllocator idAllocator = new PlanNodeIdAllocator();
 
         // plan statement
-        LogicalPlanner logicalPlanner = new LogicalPlanner(session, planOptimizers, idAllocator, metadata, sqlParser, statsCalculator, costCalculator, warningCollector);
+        LogicalPlanner logicalPlanner = new LogicalPlanner(session, planOptimizers, idAllocator, metadata, new TypeAnalyzer(sqlParser, metadata), statsCalculator, costCalculator, warningCollector);
```

Comment:
I think you can inject `TypeAnalyer` to constructor

Reply:
Similar to SqlQueryExecution, this class has all it needs and requires a SqlParser anyway to create an Analyzer, so no need to further pollute the constructor. Also, it'd be confusing to have it take a type analyzer but not an analyzer.